doc/nemo.rst

```diff
@@ -94,30 +94,23 @@ Implicit
 ^^^^^^^^
 
 The use of Fortran array notation is encouraged in the NEMO Coding
-Conventions and is employed throughout the NEMO code base. PSyclone
-therefore also recognises the loops implied by this notation. The type
-of loop is inferred from the position of the colon in the array
-subscripts. Since NEMO uses `(ji,jj,jk)` index ordering (longitude,
-latitude, levels) this means that loop types are determined
-according to the following table:
+Conventions :cite:`nemo_code_conv` (section 4.2) and is employed
+throughout the NEMO code base. The Coding Conventions mandate that the
+shape of every array in such expressions must be specified, e.g.::
 
-===========  ===============  ===========
-Array index  Loop type        Loop limits
-===========  ===============  ===========
-1            Longitude        1, jpi
-2            Latitude         1, jpj
-3            Vertical levels  1, jpk
-===========  ===============  ===========
+    onedarraya(:) = onedarrayb(:) + onedarrayc(:)
+    twodarray (:,:) = scalar * anothertwodarray(:,:)
 
-Currently PSyclone will convert every implicit loop it encounters in
-one of these dimensions into an explicit loop over the full
-sub-domain. This is because it has been found by CMCC and Intel that
-the Intel compiler is better able to OpenMP-parallelise explicit loops.
-However, as with all compiler-specific optimisations, this situation
-may change and will depend on the compiler being used.
+PSyclone therefore also recognises the loops implied by this
+notation.
 
-If an implicit loop is encountered in an array index other than 1-3 then
-currently PSyclone will raise an error.
+It has been found by CMCC and Intel that the Intel compiler is better
+able to OpenMP-parallelise explicit loops.  Therefore, PSyclone
+provides a transformation to allow the user to convert implicit loops
+into explicit loop nests (see :ref:`nemo-transformations`). However,
+as with all compiler-specific optimisations, whether or not this
+transformation is beneficial will depend on the precise details of the
+compiler being used.
```

Comment:
I think it makes sense to add a note about the current implementation being conservative about funding implicit loops and that in future work this will be improved, as it is information that the user would want to know  e.g.
Note, not all uses of Fortran array notation in NEMO imply a loop. For instance, xxx. At this time all implicit loops with xxx structure are not transformed. Future versions will xxx.

Reply:
Done.